p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -747,35 +747,6 @@ private boolean checkSignature(PublicKey ownerPubKey, byte[] hashOfDataAndSeqNr,
         }
     }
 
-    private boolean checkIfStoredDataPubKeyMatchesNewDataPubKey(PublicKey ownerPubKey, ByteArray hashOfData) {
-        ProtectedStorageEntry storedData = map.get(hashOfData);
-        boolean result = storedData.getOwnerPubKey() != null && storedData.getOwnerPubKey().equals(ownerPubKey);
-        if (!result)
-            log.warn("New data entry does not match our stored data. storedData.ownerPubKey=" +
-                    (storedData.getOwnerPubKey() != null ? storedData.getOwnerPubKey().toString() : "null") +
-                    ", ownerPubKey=" + ownerPubKey);
-
-        return result;
-    }
-
-    private boolean checkIfStoredMailboxDataMatchesNewMailboxData(PublicKey receiversPubKey, ByteArray hashOfData) {
-        ProtectedStorageEntry storedData = map.get(hashOfData);
-        if (storedData instanceof ProtectedMailboxStorageEntry) {
-            ProtectedMailboxStorageEntry entry = (ProtectedMailboxStorageEntry) storedData;
-            // publicKey is not the same (stored: sender, new: receiver)
-            boolean result = entry.getReceiversPubKey().equals(receiversPubKey)
-                    && get32ByteHashAsByteArray(entry.getProtectedStoragePayload()).equals(hashOfData);
```

Comment:
you did not migrate the hash calculation and comparison to the `ProtectedMailboxStorageEntry`. Can you elaborate?

Reply:
Absolutely. This one was really confusing at first and I'm happy to help explain so you can verify it is correct.

By renaming some variables it makes it a bit easier to understand:

```
private boolean checkIfStoredMailboxDataMatchesNewMailboxData(PublicKey removeEntryReceiversPubKey, ByteArray removeEntryPayloadHash) {	
        ProtectedStorageEntry existingEntry = map.get(removeEntryPayloadHash);	
        if (storedData instanceof ProtectedMailboxStorageEntry) {	
            ProtectedMailboxStorageEntry existingMailboxEntry = (ProtectedMailboxStorageEntry) existingEntry;	
            // publicKey is not the same (stored: sender, new: receiver)	
            boolean result = existingMailboxEntry.getReceiversPubKey().equals(removeEntryReceiversPubKey)	
                    && get32ByteHashAsByteArray(existingMailboxEntry.getProtectedStoragePayload()).equals(removeEntryPayloadHash);
```

So, it was verifying that the new removeEntry's payload has the same hash as the existingEntry's payload. But, the only way that we can get to this branch in the code is if the payload hashes are equal. Line 762 grabs the existingEntry based on the hash of the removeEntry's payload so they already must be identical because the existingEntry was inserted into the map based on its payload.